observatory-platform/observatory/platform/utils/test_utils.py

```diff
@@ -243,18 +252,31 @@ def assert_gcp_dependencies(self):
 
         assert self.create_gcp_env, "Please specify the Google Cloud project_id and data_location"
 
-    def add_bucket(self) -> str:
+    def add_bucket(self, prefix: str = "") -> str:
         """Add a Google Cloud Storage Bucket to the Observatory environment.
 
         The bucket will be created when create() is called and deleted when the Observatory
         environment is closed.
 
+        :param prefix: an optional additional prefix for the bucket.
         :return: returns the bucket name.
         """
 
         self.assert_gcp_dependencies()
-        bucket_name = random_id()
-        self.buckets.append(bucket_name)
+        if (self.prefix != "") and (prefix != ""):
+            bucket_name = f"{self.prefix}_{prefix}_{random_id()}"
+        elif self.prefix != "" and (prefix == ""):
+            bucket_name = f"{self.prefix}_{random_id()}"
+        elif (self.prefix == "") and (prefix != ""):
+            bucket_name = f"{prefix}_{random_id()}"
+        else:
+            bucket_name = random_id()
```

Comment:
This is an alternative way to make the prefix which might be easier to follow (an empty string evaluates to False as does None):

```python
parts = []
if self.prefix:
  parts.append(self.prefix)
if prefix:
  parts.append(prefix)
parts.append(random_id())
bucket_name = "_".join(parts)
```